### ISSUE TITLE
[ITK 5.1] Unresolved external symbols creating ITKIOImageBasePython

### ISSUE THREAD

**Issue by Neumann-A** [InsightSoftwareConsortium/ITK#1803]
### Description
```
Wrapping\Modules\ITKIOImageBase\CMakeFiles\ITKIOImageBasePython.dir\itkImageSeriesWriterPython.cpp.obj -LIBPATH:D:\qt2\installed\x64-windows\debug\lib D:\qt2\installed\x64-windows\debug\lib\double-conversion.lib lib\itksys-5.1.lib lib\itkvnl_algo-5.1.lib lib\itkvnl-5.1.lib lib\itkv3p_netlib-5.1.lib lib\itknetlib-5.1.lib lib\itkvcl-5.1.lib lib\itkNetlibSlatec-5.1.lib D:\qt2\installed\x64-windows\debug\lib\zlibd.lib lib\ITKMetaIO-5.1.lib lib\ITKPath-5.1.lib lib\ITKLabelMap-5.1.lib lib\ITKFastMarching-5.1.lib lib\ITKPolynomials-5.1.lib lib\ITKBiasCorrection-5.1.lib lib\ITKColormap-5.1.lib lib\ITKConvolution-5.1.lib lib\ITKDICOMParser-5.1.lib lib\ITKDeformableMesh-5.1.lib lib\ITKDenoising-5.1.lib lib\ITKDiffusionTensorImage-5.1.lib D:\qt2\installed\x64-windows\debug\lib\expat.lib lib\itkgdcmDICT-5.1.lib lib\itkgdcmMSFF-5.1.lib lib\ITKznz-5.1.lib lib\ITKniftiio-5.1.lib lib\ITKgiftiio-5.1.lib lib\ITKPDEDeformableRegistration-5.1.lib lib\ITKIOBioRad-5.1.lib lib\ITKIOBruker-5.1.lib lib\ITKIOCSV-5.1.lib lib\ITKIOGE-5.1.lib D:\qt2\installed\x64-windows\lib\jpeg.lib D:\qt2\installed\x64-windows\debug\lib\jpegd.lib lib\itkopenjpeg-5.1.lib lib\ITKIOJPEG2000-5.1.lib D:\qt2\installed\x64-windows\lib\tiff.lib D:\qt2\installed\x64-windows\debug\lib\tiffd.lib D:\qt2\installed\x64-windows\lib\lzma.lib D:\qt2\installed\x64-windows\debug\lib\lzmad.lib lib\ITKIOLSM-5.1.lib lib\itkminc2-5.1.lib lib\ITKIOMINC-5.1.lib lib\ITKIOMRC-5.1.lib lib\ITKNrrdIO-5.1.lib D:\qt2\installed\x64-windows\lib\libpng16.lib D:\qt2\installed\x64-windows\debug\lib\libpng16d.lib lib\ITKIOSiemens-5.1.lib lib\ITKIOSpatialObjects-5.1.lib lib\ITKIOStimulate-5.1.lib lib\ITKTransformFactory-5.1.lib lib\ITKIOTransformHDF5-5.1.lib lib\ITKIOTransformInsightLegacy-5.1.lib lib\ITKIOTransformMatlab-5.1.lib lib\ITKKLMRegionGrowing-5.1.lib lib\itklbfgs-5.1.lib lib\ITKMarkovRandomFieldsClassifiers-5.1.lib lib\ITKQuadEdgeMeshFiltering-5.1.lib lib\ITKRegionGrowing-5.1.lib lib\ITKRegistrationMethodsv4-5.1.lib lib\ITKTestKernel-5.1.lib lib\ITKVideoIO-5.1.lib lib\ITKVtkGlue-5.1.lib lib\ITKWatersheds-5.1.lib lib\ITKFFT-5.1.lib lib\itkgdcmDICT-5.1.lib lib\itkgdcmIOD-5.1.lib lib\itkgdcmDSED-5.1.lib lib\itkgdcmCommon-5.1.lib crypt32.lib lib\itkgdcmjpeg8-5.1.lib lib\itkgdcmjpeg12-5.1.lib lib\itkgdcmjpeg16-5.1.lib lib\itkgdcmopenjp2-5.1.lib lib\itkgdcmcharls-5.1.lib rpcrt4.lib D:\qt2\installed\x64-windows\debug\lib\expat.lib lib\ITKniftiio-5.1.lib lib\ITKznz-5.1.lib lib\ITKIOIPL-5.1.lib lib\ITKIOXML-5.1.lib D:\qt2\installed\x64-windows\debug\lib\hdf5_cpp_D.lib D:\qt2\installed\x64-windows\debug\lib\hdf5_D.lib lib\ITKIOTransformBase-5.1.lib lib\ITKTransformFactory-5.1.lib lib\ITKImageFeature-5.1.lib lib\ITKOptimizersv4-5.1.lib lib\ITKOptimizers-5.1.lib lib\ITKIOBMP-5.1.lib lib\ITKIOGDCM-5.1.lib lib\ITKIOGIPL-5.1.lib lib\ITKIOJPEG-5.1.lib lib\ITKIOTIFF-5.1.lib lib\ITKIOMeshBYU-5.1.lib lib\ITKIOMeshFreeSurfer-5.1.lib lib\ITKIOMeshGifti-5.1.lib lib\ITKIOMeshOBJ-5.1.lib lib\ITKIOMeshOFF-5.1.lib lib\ITKIOMeshVTK-5.1.lib lib\ITKIOMeshBase-5.1.lib lib\ITKQuadEdgeMesh-5.1.lib lib\ITKIOMeta-5.1.lib lib\ITKMetaIO-5.1.lib D:\qt2\installed\x64-windows\debug\lib\zlibd.lib comctl32.lib wsock32.lib lib\ITKIONIFTI-5.1.lib lib\ITKIONRRD-5.1.lib lib\ITKIOPNG-5.1.lib lib\ITKIOVTK-5.1.lib lib\ITKIOImageBase-5.1.lib lib\ITKVideoCore-5.1.lib lib\ITKVTK-5.1.lib D:\qt2\installed\x64-windows\debug\lib\vtkWrappingPythonCore-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\python37_d.lib D:\qt2\installed\x64-windows\debug\lib\vtkRenderingOpenGL2-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkRenderingUI-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\glew32d.lib opengl32.lib glu32.lib D:\qt2\installed\x64-windows\debug\lib\vtkInteractionWidgets-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkInteractionStyle-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkImagingSources-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkIOImage-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkRenderingFreeType-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\freetyped.lib D:\qt2\installed\x64-windows\debug\lib\vtkImagingCore-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkRenderingCore-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkFiltersSources-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkFiltersGeneral-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkFiltersCore-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkCommonExecutionModel-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkCommonDataModel-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkCommonTransforms-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkCommonMisc-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkCommonMath-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtkCommonCore-9.0d.lib D:\qt2\installed\x64-windows\debug\lib\vtksys-9.0d.lib lib\ITKStatistics-5.1.lib lib\itkNetlibSlatec-5.1.lib lib\ITKSpatialObjects-5.1.lib lib\ITKMesh-5.1.lib lib\ITKTransform-5.1.lib lib\ITKPath-5.1.lib lib\ITKCommon-5.1.lib lib\itksys-5.1.lib ws2_32.lib dbghelp.lib psapi.lib lib\ITKVNLInstantiation-5.1.lib lib\itkvnl_algo-5.1.lib lib\itkvnl-5.1.lib lib\itkv3p_netlib-5.1.lib lib\itknetlib-5.1.lib lib\itkvcl-5.1.lib lib\ITKSmoothing-5.1.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib 
   Creating library lib\ITKIOImageBasePython.lib and object lib\ITKIOImageBasePython.exp
itkImageFileReaderPython.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) void __cdecl itk::HDF5ImageIOFactoryRegister__Private(void)" (__imp_?HDF5ImageIOFactoryRegister__Private@itk@@YAXXZ) referenced in function "void __cdecl itk::`anonymous namespace'::`dynamic initializer for 'ImageIOFactoryRegisterRegisterList''(void)" (??__EImageIOFactoryRegisterRegisterList@?A0xbb56962a@itk@@YAXXZ)
itkImageFileWriterPython.cpp.obj : error LNK2001: unresolved external symbol "__declspec(dllimport) void __cdecl itk::HDF5ImageIOFactoryRegister__Private(void)" (__imp_?HDF5ImageIOFactoryRegister__Private@itk@@YAXXZ)
itkImageSeriesReaderPython.cpp.obj : error LNK2001: unresolved external symbol "__declspec(dllimport) void __cdecl itk::HDF5ImageIOFactoryRegister__Private(void)" (__imp_?HDF5ImageIOFactoryRegister__Private@itk@@YAXXZ)
itkImageSeriesWriterPython.cpp.obj : error LNK2001: unresolved external symbol "__declspec(dllimport) void __cdecl itk::HDF5ImageIOFactoryRegister__Private(void)" (__imp_?HDF5ImageIOFactoryRegister__Private@itk@@YAXXZ)
lib\_ITKIOImageBasePython.pyd : fatal error LNK1120: 1 unresolved externals
```
On which file need I to run dumpbin on to figure out whats wrong with the symbols?

**Comment by Neumann-A** [comment]
ok that link line is missing ITKIOHDF5-5.1.lib but where should i add that?

**Comment by Neumann-A** [comment]
closing this for the meantime until I found the proper fix for #1799
